geonode/api/resourcebase_api.py

```diff
@@ -525,6 +526,25 @@ class Meta(CommonMetaApi):
         resource_name = 'maps'
 
 
+class MapLayersResource(CommonModelApi):
+
+    """Includes everything from Maps API plus details for each layer in map"""
+
+    class Meta(CommonMetaApi):
+        queryset = Map.objects.distinct().order_by('-date')
+        if settings.RESOURCE_PUBLISHING:
+            queryset = queryset.filter(is_published=True)
+        resource_name = 'maplayers'
+
+    def dehydrate(self, bundle):
```

Comment:
@cstephen did you do any performance test on this api? Dehydrate loops over each result and for each result you are looping over each maplayer. The performances may decrease quickly with the grow of the number of maps.